amundsen_application/static/js/components/TableDetail/ExploreButton/tests/index.spec.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import ExploreButton, { ExploreButtonProps } from 'components/TableDetail/ExploreButton';
+import { TableMetadata } from 'interfaces/TableMetadata';
+import AppConfig from 'config/config';
+import { logClick } from 'ducks/utilMethods';
+
+describe('ExploreButton', () => {
+  const setup = (tableDataOverrides?: Partial<TableMetadata>) => {
+    const props = {
+      tableData: {
+        badges: [],
+        cluster: 'cluster',
+        columns: [],
+        database: 'database',
+        is_editable: false,
+        is_view: false,
+        key: '',
+        schema: 'schema',
+        table_name: 'table_name',
+        table_description: '',
+        table_writer: { application_url: '', description: '', id: '', name: '' },
+        partition: { is_partitioned: true },
+        table_readers: [],
+        source: { source: '', source_type: '' },
+        watermarks: [],
+        ...tableDataOverrides,
+      },
+    };
+    const wrapper = shallow<ExploreButton>(<ExploreButton {...props } />)
+    return { props, wrapper };
+  };
+
+  describe('generateUrl', () => {
+    const { props, wrapper } = setup();
+    const urlGeneratorSpy = jest.spyOn(AppConfig.tableProfile, "exploreUrlGenerator");
+
+    it('calls url generator with the partition value and key, if partitioned', () => {
+      urlGeneratorSpy.mockClear();
+      const tableData = {
+        ...props.tableData,
+        partition: {
+          is_partitioned: true,
+          key: 'partition_key',
+          value: 'partition_value',
+        },
+      };
+      const partition = tableData.partition;
+      wrapper.instance().generateUrl(tableData);
+      expect(urlGeneratorSpy).toHaveBeenCalledWith(
+        tableData.database,
+        tableData.cluster,
+        tableData.schema,
+        tableData.table_name,
+        partition.key,
+        partition.value);
+    });
+
+
+    it('calls url generator with no partition value and key, if not partitioned', () => {
+      urlGeneratorSpy.mockClear();
+      const tableData = {
+        ...props.tableData,
+        partition: {
+          is_partitioned: false,
+        },
+      };
+      wrapper.instance().generateUrl(tableData);
+      expect(urlGeneratorSpy).toHaveBeenCalledWith(
+        tableData.database,
+        tableData.cluster,
+        tableData.schema,
+        tableData.table_name);
+    });
+  });
+
+
+  describe('render', () => {
+    beforeEach(() => {
```

Comment:
nit: In the future let's mock the appropriate config utils to cut down on tech debt because eventually we should modify all the tests that directly modify the imported `AppConfig1. Example: https://github.com/lyft/amundsenfrontendlibrary/blob/master/amundsen_application/static/js/components/common/SearchBar/InlineSearchResults/SearchItemList/tests/index.spec.tsx#L12-L13

The mock implementations for each unit tests can also then even leverage `mockImplementationOnce()` to make sure that the mocked value only applies to that test.